cmd/cluster-agent/admissioncontroller/server.go

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// Package admissioncontroller implements the agent admission webhook
+// endpoint. This server receives k8s objects from the cluster and
+// preforms changes or admission control to such objects.
+//
+// More info: https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/
+package admissioncontroller
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"net"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	server *http.Server
+)
+
+func getListener(port int) (net.Listener, error) {
+	return net.Listen("tcp", fmt.Sprintf("0.0.0.0:%v", port))
+}
+
+// StartServer creates the router and starts the HTTP server
+func StartServer() error {
+	certFile := config.Datadog.GetString("cluster_agent.admission_controller.tls_cert_file")
+	keyFile := config.Datadog.GetString("cluster_agent.admission_controller.tls_key_file")
+
+	log.Debugf("loading TLS certificate: cert.pem %s, key.pem %s", certFile, keyFile)
+
+	pair, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		log.Errorf("failed to load key pair: %v", err)
```

Comment:
```suggestion
		log.Errorf("Failed to load key pair: %v", err)
```
Please capitalize all log messages - this is something we've done everywhere in this repository.

Reply:
I've changed them all to *not* capitalized because of an earlier comment in this PR, I'll revert now 😅

Reply:
That was also me. Let me try to explain:
* Log messages: these get displayed in the console when you look at the log output of the program, so they need to be capitalised because it's a full sentence.
* Error messages: these do not get capitalised because they are not full sentences and they will be appended to other sentences. For example, you might return an error from a function like `"can not open file"`. The caller of this function might then log `"Error starting program: can not open file"`. Capitalising the error here would make this much less user friendly. This can also happen as a chain of multiple errors sometimes.

Returned error messages start with lower case, but logged messages start with uppercase. I hope this makes sense. I think the confusion was around the part that the logged message was at the error level, and you were also returning its value, which was wrong because of previously explained reasons. Understandable 🙂